src/Sylius/Bundle/ApiBundle/spec/Validator/Constraints/CorrectAddressOrderValidatorSpec.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Bundle\ApiBundle\Command\Cart\AddItemToCart;
+use Sylius\Bundle\ApiBundle\Command\Checkout\AddressOrder;
+use Sylius\Bundle\ApiBundle\Command\Checkout\CompleteOrder;
+use Sylius\Bundle\ApiBundle\Validator\Constraints\AddingEligibleProductVariantToCart;
+use Sylius\Bundle\ApiBundle\Validator\Constraints\CorrectAddressOrder;
+use Sylius\Bundle\ApiBundle\Validator\Constraints\CorrectAddressOrderValidator;
+use Sylius\Component\Core\Model\AddressInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidatorInterface;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+
+final class CorrectAddressOrderValidatorSpec extends ObjectBehavior
+{
+    function it_is_a_constraint_validator(): void
+    {
+        $this->shouldImplement(ConstraintValidatorInterface::class);
+    }
+
+    function it_throws_an_exception_if_value_is_not_an_instance_of_address_order_command(): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('validate', [new CompleteOrder(), new AddingEligibleProductVariantToCart()]);
```

Comment:
You should pass here a correct constraint
```suggestion
            ->during('validate', [new CompleteOrder(), new CorrectAddressOrder()]);
```